doc/CONTRIBUTING.md

```diff
@@ -31,13 +31,20 @@ php-cs-fixer fix --verbose
 
 #### The content
 
-A Pull Request should concern one and **only one** subject.
+Ideally, a Pull Request should concern one and **only one** subject, so that it
+remains clear, and independant changes can be merged quickly.
 
-If you want to fix a typo and improve the performance of a process, you have to do two **separated** PR.
+If you want to fix a typo and improve the performance of a process, you should
+try as much as possible to it in a **separate** PR, so that we can quickly
+merge one while discussing the other.
 
 The goal is to have a clear commit history and make possible revert easier.
 
-If you found an issue/typo while writing your change that is not related to your work, please do another PR for that.
+If you found an issue/typo while writing your change that is not related to
+your work, please do another PR for that. If however, you fall in an edge case
+where you can't easily do that (if the typo/issue is on a line you want to
+change), you will probably need to do something more complicated. In that case,
+please refer to the [Composing a PR](./composing-a-pr.md) document.
```

Comment:
I would like to have you're document later on another PR directly on the main repo so we can discuss on it with the others.

Until that, can you change the sentence beginning by "If however" by:

> In some **rare** case, you probably might have to do it on the same PR. If you have a good justification, feel free to do it and explain it.

(with a better english if you want :stuck_out_tongue:).